orca-kayenta/src/main/groovy/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryPipelineStage.groovy

```diff
@@ -40,4 +48,20 @@ class RunCanaryPipelineStage implements StageDefinitionBuilder {
   String getType() {
     STAGE_TYPE
   }
+
+  @Override
+  CancellableStage.Result cancel(Stage stage) {
+    Map<String, Object> context = stage.getContext()
+    String canaryPipelineExecutionId = (String)context.get("canaryPipelineExecutionId")
+
+    log.info("Cancelling stage (stageId: $stage.id: executionId: $stage.execution.id, canaryPipelineExecutionId: $canaryPipelineExecutionId, context: $stage.context)")
+
+    try {
+      kayentaService.cancelPipelineExecution(canaryPipelineExecutionId, "")
+    } catch (Exception e) {
+      log.info("Failed to cancel stage (stageId: $stage.id, executionId: $stage.execution.id), e: $e.message")
+    }
+
+    return new CancellableStage.Result(stage, [:])
```

Comment:
Is it normal for a result to look like this, with only the stage and no surrounding context? Do you get success/failure from looking at whether the stage was cancelled? I guess there is no error handling so reasons for failure to cancel dont matter.

If the caller just calls cancel and doesnt care about what happens after that then this is ok. Otherwise it seems odd.

LGTM otherwise.

Reply:
Yeah, and I took this style from an existing cancellable stage.

Reply:
Seems like the majority of cancellable stages don't return any additional context.